file_catalog/schema/validation.py

```diff
@@ -88,57 +89,69 @@ def _find_all_field_vals(metadata: types.Metadata, fields: List[str]) -> Dict[st
-            if self._field_vals_are_different(field, val, old_metadata):
-                apihandler.send_error(
-                    400,
-                    reason=f"Validation Error: forbidden attribute creation '{field}'",
-                    file=apihandler.files_url,
-                )
-                return True
-        return False
-
-    def has_forbidden_attributes_modification(self, apihandler: Any, metadata: types.Metadata, old_metadata: types.Metadata) -> bool:
-        """Check if `metadata` has forbidden attribute updates."""
-        forbidden_matches = self._find_all_field_vals(metadata, self.FORBIDDEN_FIELDS_UPDATE)
+    @staticmethod
+    def _has_forbidden_attributes(
+        apihandler: Any,
+        metadata: types.Metadata,
+        old_metadata: types.Metadata,
+        forbidden_attributes: List[str],
+        http_error_message: str,
+    ) -> bool:
+        forbidden_matches = Validation._find_all_field_vals(
+            metadata, forbidden_attributes
+        )
 
         for field, val in forbidden_matches.items():
-            if self._field_vals_are_different(field, val, old_metadata):
+            if Validation._field_vals_are_different(field, val, old_metadata):
                 apihandler.send_error(
                     400,
-                    reason=f"Validation Error: forbidden attribute update '{field}'",
+                    reason=f"Validation Error: {http_error_message} '{field}'",
                     file=apihandler.files_url,
                 )
                 return True
         return False
 
-    def validate_metadata_creation(self, apihandler: Any, metadata: types.Metadata) -> bool:
-        """Validate metadata for creation.
+    def has_forbidden_attributes_creation(
+        self, apihandler: Any, metadata: types.Metadata, old_metadata: types.Metadata
+    ) -> bool:
+        """Check if `metadata` has forbidden attributes and whether they have changed.
 
-        Utilizes `send_error` and returns `False` if validation failed.
-        If validation was successful, `True` is returned.
+        Returns `True` if it has forbidden attributes.
         """
-        if self.has_forbidden_attributes_creation(apihandler, metadata, {}):
-            return False
-        return self.validate_metadata_modification(apihandler, metadata)
+        return self._has_forbidden_attributes(
+            apihandler,
+            metadata,
+            old_metadata,
+            self.FORBIDDEN_FIELDS_CREATION,
+            "forbidden attribute creation",
+        )
+
+    def has_forbidden_attributes_modification(
+        self, apihandler: Any, metadata: types.Metadata, old_metadata: types.Metadata
+    ) -> bool:
+        """Check if `metadata` has forbidden attribute updates."""
+        return self._has_forbidden_attributes(
+            apihandler,
+            metadata,
+            old_metadata,
+            self.FORBIDDEN_FIELDS_MODIFICATION,
+            "forbidden attribute update",
```

Comment:
You changed the name from update to modification, did you want to change this string too?

Reply:
sure